devicetypes/qubino/qubino-flush-dimmer.src/qubino-flush-dimmer.groovy

```diff
@@ -0,0 +1,890 @@
+			map.name = "carbonDioxide"
+			switch (cmd.event) {
+				case 0x02:
+					map.value = "detected"
+					map.descriptionText = "$device.displayName detected CO2"
+					break
+				case 0x00:
+					map.value = "clear"
+					map.descriptionText = "$device.displayName detected no CO2"
+					break
+			}
+		} else if (cmd.notificationType == 0x04) { // Overheat Alarm
+			map.name = "temperatureAlarm"
+			switch (cmd.event) {
+				case 0x00:
+					map.value = "cleared"
+					map.descriptionText = "$device.displayName heat is clear"
+					break
+				case 0x02:
+					map.value = "heat"
+					map.descriptionText = "$device.displayName heat was detected!"
+					break
+			}
+		} else if (cmd.notificationType == 0x05) { // Water Alarm
+			map.name = "water"
+			switch (cmd.event) {
+				case 0x02:
+					map.value = "wet"
+					map.descriptionText = "$device.displayName detected water"
+					break
+				case 0x00:
+					map.value = "dry"
+					map.descriptionText = "$device.displayName detected no water"
+					break
+			}
+		} else if (cmd.notificationType == 0x07) { // Home Security
+			map.name = "motion"
+			if (cmd.event == 0x00) {
+				map.value = "inactive"
+				map.descriptionText = "$device.displayName detected no motion"
+			} else if (cmd.event == 0x07 || cmd.event == 0x08) {
+				map.value = "active"
+				map.descriptionText = "$device.displayName detected motion"
+			}
+		}
+	} else if (cmd.hasProperty("sensorType")) {
+		componentLabel = "Qubino Temperature Sensor"
+		childDTHname = "Child Temperature Sensor"
+		switch (cmd.sensorType) {
+			case 0x01:
+				log.debug ("cmd: ${cmd}")
+				map.name = "temperature"
+				def cmdScale = cmd.scale == 1 ? "F" : "C"
+				map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, cmd.precision)
+				map.unit = getTemperatureScale()
+				break
+		}
+	}
+	childDni = "${device.deviceNetworkId}-${childUniqueDni[componentLabel]}"
+
+	if (doesChildDeviceExist(componentLabel)) {
+		log.debug("Found device: ${componentLabel}")
+	} else {
+		log.debug ("Device ${componentLabel} not found. Creating new child device.")
+		createChildDevice(componentLabel, childDTHname, childDni)
+	}
+	sendEventToChild(map, childDni)
+}
+
+def sendEventToChild(event, childDni) {
+	def child = childDevices.find { it.deviceNetworkId == childDni }
+	log.debug "Sending event: ${event} to child: ${child}"
+	child?.sendEvent(event)
+}
+
+private refreshChild() {
+	sendHubCommand(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1, scale: 0).format())
+}
+
+def doesChildDeviceExist(componentLabel) {
+	def exists = (childDevices.find{it.toString() == componentLabel.toString()} == null) ? false : true
+}
+
+def getAppSwitchOnOffTime() {
+	/*
+	User can set on/off time between 50 - 255 ms only when the dimmer is activated from the physical wall-switch.
+	This function adjusts that time to be used also when activating from the Mobile App (1-3s).
+	*/
+	def dimmingTimePref = parameterMap.find({it.parameterNumber == 65})

```

Comment:
I would change "65" to a constant to avoid magic numbers in a code (it refers to all similar occurrences)

Reply:
@KKlimczukS Maybe I will go for something like:  parameterMap.find( { it.key == 'dimmingTimeSoftOnOff' }) .